### PR TITLE
add keyball

### PR DESCRIPTION
**Pull request by Yowkees**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Yowkees** [comment]
I created new keyboard "keyball".
Thank you very much for your help. Thank you.

File: keyboards/keyball/rev1/rev1.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    L10, L11, L12, L13, L14, L15,           R10, R11, R12, R13, R14, R15, \
+    L20, L21, L22, L23, L24, L25,           R20, R21, R22, R23, R24, R25, \
+    L30, L31,    L32, L33, L34, L35,      R30, R31, R32,   R33, R34, R35 \
+    ) \
+    LAYOUT( \
+        KC_##L00, KC_##L01, KC_##L02, KC_##L03, KC_##L04, KC_##L05,                     KC_##R00, KC_##R01, KC_##R02, KC_##R03, KC_##R04, KC_##R05, \
+        KC_##L10, KC_##L11, KC_##L12, KC_##L13, KC_##L14, KC_##L15,                     KC_##R10, KC_##R11, KC_##R12, KC_##R13, KC_##R14, KC_##R15, \
+        KC_##L20, KC_##L21, KC_##L22, KC_##L23, KC_##L24, KC_##L25,                     KC_##R20, KC_##R21, KC_##R22, KC_##R23, KC_##R24, KC_##R25, \
+        KC_##L30, KC_##L31, KC_##L32, KC_##L33, KC_##L34, KC_##L35,                     KC_##R30, KC_##R31, KC_##R32, KC_##R33, KC_##R34, KC_##R35 \
+    )
```
Comment (drashna) [inline keyboards/keyball/rev1/rev1.h:33]: Please remove the `LAYOUT_kc` macro.  it is deprecated and we don't support the usage of it 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
Most of the pointing device code could be moved to the keyboard level, and ... in fact, should be. 

If you take a look at the ploopyco stuff, you can see how I've done this, and maintained the ability for users to customize the functionality of the sensor's behavior. 

Additionally, if you check out `keyboards/handwired/dactyl_manuform/5x6_right_trackball/`, I have a split keyboard that the sensor works, regardless of which side the keyboard is plugged into.

**Comment by drashna** [comment]
That said, awesome to see another split with a trackball!!

File: keyboards/keyball/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+		optical_sensor_init();
+	}
+}
+
+void keyboard_post_init_user() {
+    debug_enable = true;
+    debug_mouse = true;
+}
+
+void matrix_scan_user(void) {
```
[the remark this answers is not in the document]
Comment (Yowkees) [reply keyboards/keyball/keymaps/via/keymap.c:172]: Thank you very much for your checking. I will fix it later. Thank you!!

**Pull request closed without merging**